### ISSUE TITLE
[SPIR-V] GLSL mix() emits scalar lerp parameter

### ISSUE THREAD

**Issue by Themaister** [KhronosGroup/glslang#99]
```
#version 310 es
precision mediump float;

out vec4 FragColor;
in vec4 v0;
in vec4 v1;

void main()
{
    FragColor = mix(v0, v1, 0.9);
}
```

Compiles to
```
              15:    6(float) Constant 1063675494
         4(main):           2 Function None 3
               5:             Label
              12:    7(fvec4) Load 11(v0)
              14:    7(fvec4) Load 13(v1)
              16:    7(fvec4) ExtInst 1(GLSL.std.450) 46(Mix) 12 14 15
```

Scalar lerp parameter is legal GLSL afaik, but not GLSL.std.450.

**Comment by johnkslang** [comment]
Correct, the scalar should be explicitly widened to a vector.

**Comment by johnkslang** [comment]
Fixed by e7c83cfb8c35b8bd3a9e39b619c4a8618f8f80f4, along with all of min, max, clamp, mix, step, and smoothstep.
